### PR TITLE
Youtube Integration

### PR DESCRIPTION
**Pull request by conrad-vanl**
## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-api/src/data/willow-tv/data-source.js [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
+  willSendRequest = (request) => {
+    request.params.set('key', process.env.YOUTUBE_API_KEY);
+  };
 
-  async getLiveStream() {
-    return this.get('live');
-  }
+  async getFromId(id) {
+    const result = await this.get('videos', { part: 'snippet', id });
+    console.log(result.items[0]);
```
Comment (vinnyjth) [inline packages/apollos-api/src/data/willow-tv/data-source.js]:
```suggestion
```

File: packages/apollos-api/src/data/willow-tv/data-source.js [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
-export default class LiveStream extends RESTDataSource {
-  baseURL = 'https://willowcreek.tv/api/';
+export default class Youtube extends RESTDataSource {
+  baseURL = 'https://content.googleapis.com/youtube/v3';
 
-  async getAll() {
-    return this.get('all', null, { ttl: 86400 });
-  }
+  willSendRequest = (request) => {
+    request.params.set('key', process.env.YOUTUBE_API_KEY);
```
Comment (vinnyjth) [inline packages/apollos-api/src/data/willow-tv/data-source.js]: I've been trying to avoid leaving configuration that happens via `process.env`, and moving it to the Config.yml (but referencing a ENV variable) could we do that here? 

I think it's more self documenting when we do that :)

File: packages/apollos-api/config.yml [hunk 3]
```diff
@@ -27,6 +27,9 @@ TWILIO:
   ACCOUNT_SID: ${TWILIO_ACCOUNT_SID}
   AUTH_TOKEN: ${TWILIO_AUTH_TOKEN}
   FROM_NUMBER: ${TWILIO_FROM_NUMBER}
+YOUTUBE:
+  API_KEY: ${YOUTUBE_API_KEY}
+  PLAYLIST_ID: 'UUmFRqL-BDXeaYwMxYDfhu_w'
```
Comment (conrad-vanl) [inline packages/apollos-api/config.yml]: Before this is merged I want to move this from using a single playlist to playlists for each campus. I'm still thinking this might be in Rock, so going to leave this open for now.

Comment (vinnyjth) [reply packages/apollos-api/config.yml]: Solution here which we can build off later
https://github.com/Differential/willow-prototype/pull/7/commits/6a23cb12982a4e82ec92efcc4367263784f89f8e
